### PR TITLE
PostgreSQL 14 upgrade

### LINKED ISSUE #4351: Upgrade Postgresql major version to 14
Our current version major version is 13. We'll need to look into the migration efforts for our Azure Databases. https://github.com/CDCgov/prime-simplereport/pull/4350

### PR DESCRIPTION
**Pull request by alismx**
# DATABASE/DEVOPS PULL REQUEST

# DO NOT MERGE UNTIL DATABASE UPGRADES ARE COMPLETE

## Related Issue

- resolves #4351 

## Changes Proposed

- Set db version to 14.

## Additional Information

- Azure does prod db backups around 9:30 Pacific
- Wait for latest backup to prod, then run upgrade, when upgrades are complete in all environments, merge this PR.

## Testing

- How should reviewers verify this PR?

## Upgrade

Here's the updated checklist with the items from the array added:

### Create a backup instance
- [x] Create a new instance of the database in the same environment.
    - [x] demo
    - [x] dev
    - [x] dev2
    - [x] dev3
    - [x] dev4
    - [x] dev5
    - [x] dev6
    - [x] dev7
    - [x] pentest
    - [x] test
    - [x] training
    - [x] stg
    - [ ] prod

### Upgrade to PostgreSQL 14
- [x] Confirm that the installation process for PostgreSQL 14 has been completed successfully.
    - [x] demo
    - [x] dev
    - [x] dev2
    - [x] dev3
    - [x] dev4
    - [x] dev5
    - [x] dev6
    - [x] dev7
    - [x] pentest
    - [x] test
    - [x] training
    - [x] stg
    - [x] prod

### Smoke test the environment
- [x] Verify that the apps function and no critical bugs or errors are encountered.
    - [x] demo
    - [x] dev
    - [x] dev2
    - [x] dev3
    - [x] dev4
    - [x] dev5
    - [x] dev6
    - [x] dev7
    - [x] pentest
    - [x] test
    - [x] training
    - [x] stg
    - [ ] prod

### Run an adhoc Terraform Plans `alis/4351-postgresql-upgrade`
- [x] Execute terraform plan to verify the upgrade applied correctly.
- [x] Make sure the plan doesn't want make any changes to the database.
    - [x] demo
    - [x] dev
    - [x] dev2
    - [x] dev3
    - [x] dev4
    - [x] dev5
    - [x] dev6
    - [x] dev7
    - [x] pentest
    - [x] test
    - [x] training
    - [x] stg
    - [x] prod

For each of the following environments, make sure that the above checklist has been completed.
- [x] demo
- [x] dev
- [x] dev2
- [x] dev3
- [x] dev4
- [x] dev5
- [x] dev6
- [x] dev7
- [x] pentest
- [x] test
- [x] training
- [x] stg
- [x] prod

### Ready for merge?
- [x] ONLY AFTER ALL THE ABOVE STEPS HAVE BEEN COMPLETED ON ALL ENVIRONMENTS, merge the PR.

## Checklist for Primary Reviewer

- [ ] Only database changes are included in this PR
- [ ] Any new tables or columns that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
- [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views (including re-granting permission to the no-PHI user if need be)
- [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] Rollback has been verifed locally and in a deployed environment
- [ ] Any changes to the startup configuration have been documented in the README

### Infrastructure
- [ ] Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### Cloud
- [ ] Oncall has been notified if this change is going in after-hours
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### Documentation
- [ ] Any changes to the startup configuration have been documented in the README

### CODE REVIEW

**Review by rin-skylight — APPROVED** [review]
Thank you so much for all of your hard work on this! 🥳

**Review by rin-skylight — APPROVED** [review]

File: .github/workflows/e2eRemote.yml [hunk 1]
```diff
@@ -32,21 +32,11 @@ on:
 
 jobs:
   
-  cypress-remote-test:
+  cypress_remote_test:
```
Comment (emyl3) [inline .github/workflows/e2eRemote.yml:35]: do we actually use this 🤔

**Review by emyl3 — APPROVED** [review]
Left one question (non-blocking) LGTM!

File: .github/workflows/e2eRemote.yml (hunk 1 above)
Comment (alismx) [reply .github/workflows/e2eRemote.yml:35]: We don't at the moment! I have every intention of enabling this job in the coming months. :smile:

Comment (emyl3) [reply .github/workflows/e2eRemote.yml:35]: Oh!! Got it thank you for the clarification!! 🎉

**Pull request merged**